adagrams/game.py

```diff
@@ -1,11 +1,145 @@
+    # this part of the code can be replaced by import counter
+    def count_letters(letters):
+        counts = {}
+        for letter in letters:
+            if letter in counts:
+                counts[letter] += 1
+            else:
+                counts[letter] = 1
+        return counts
+    
+    # convert word and letter_bank to lowercase to handle case sensitiveness
+    word = word.lower()
+    letter_bank = [letter.lower() for letter in letter_bank]
+    
+    # count the frequencies of the letters in word and letter_bank (word represents the user input words and letter_bank represent the letters that have been drawn)
+    word_counts = count_letters(word)
+    letter_bank_counts = count_letters(letter_bank)
+
+    # check to see if every letters in word is available in letter_bank
+    for letter, count in word_counts.items():
+        if letter_bank_counts.get(letter, 0) < count:
+            return False           # if the letter count in word is more than what we have in letter_bank
+    return True                    # if all the letters are available in letter_bank   
+
 
 def score_word(word):
-    pass
+    # score map, value of each letters
+    letter_scores = {
+        'A': 1, 'E': 1, 'I': 1, 'O': 1, 'U': 1,
+        'L': 1, 'N': 1, 'R': 1, 'S': 1, 'T': 1,
+        'D': 2, 'G': 2,
+        'B': 3, 'C': 3, 'M': 3, 'P': 3,
+        'F': 4, 'H': 4, 'V': 4, 'W': 4, 'Y': 4,
+        'K': 5,
+        'J': 8, 'X': 8,
+        'Q': 10, 'Z': 10
+    }
+    # convert the word to uppercase to match the key in letter_score
+    word = word.upper()
+    # initialize base score
+    base_score = 0
+    for letter in word:
+        base_score += letter_scores.get(letter, 0)
+
+    # calculate the bonus points for the words od length 7 or more
+    if len(word) >= 7:
+        bonus_points = 8
+    else: 
+        bonus_points = 0
+
+    total_score = base_score + bonus_points
+
+    return total_score
+
 
 def get_highest_word_score(word_list):
-    pass
\ No newline at end of file
+    highest_word = ""
+    highest_score = 0
+
+# loop through the word_list to get the score
+    for word in word_list: 
+        score = score_word(word)
+
+        if score > highest_score:
+            highest_word = word
+            highest_score = score
+        # check if current word's score is equal to the highest score.
+        elif score == highest_score:
+            # check if the current word has 10 characters, it will replacw the previous highest word
+            if len(word) == 10 and len(highest_word) != 10:
+                highest_word = word
+            # if the current word has less characters it will replace the previous highest word
+            elif len(word) < len(highest_word) and len(highest_word) != 10:
+                highest_word = word
+            # assign current word to highest word if the lengths of the current word and highest word are the same
+            elif len(word) == len(highest_word) and highest_word == "":
```

Comment:
What the comment says this if statement is doing is very different than what the if statement is actually doing! You've included a check to see if highest word doesn't actually exist. The only way this would ever be triggered is if the list is just empty strings, which we know isn't the case, so this final elif is actually redundant. Other than that, the logic you have here is great!